src/shims/unix/linux/epoll.rs

```diff
@@ -153,6 +153,60 @@ impl EpollInterestTable {
     }
 }
 
+impl<'tcx> EvalContextExtPriv<'tcx> for crate::MiriInterpCx<'tcx> {}
+trait EvalContextExtPriv<'tcx>: crate::MiriInterpCxExt<'tcx> {
+    /// For a specific unique file descriptor id, get its ready events and update
+    /// the corresponding ready list. This function is called whenever a file description
+    /// is registered with epoll, or when read, write, or close operations are performed,
+    /// regardless of any changes in readiness.
+    ///
+    /// This is an internal helper function and is typically not meant to be used directly.
+    /// `FileDescriptionRef::check_and_update_readiness` should be preferred.
+    fn check_and_update_readiness(
```

Comment:
Can't we inline this into FileDescriptionRef::check_and_update_readiness now?

Reply:
Yeah we can... I guess there's no good reason to keep the previous code structure.